src/client/event_handler.rs

```diff
@@ -484,6 +496,20 @@ event_handler! {
 pub trait RawEventHandler: Send + Sync {
     /// Dispatched when any event occurs
     async fn raw_event(&self, _ctx: Context, _ev: Event) {}
+
+    /// Check if `event` should be dispatched (`true`) or ignored (`false`) by the bot.
+    /// This affects [`crate::collector::collect`], [`crate::framework::Framework::dispatch`] and
+    /// this EventHandler trait.
+    ///
+    /// WARNING: This method will be run synchronously for every received event in the main dispatch
+    /// loop of the receiving shard! It should be as fast as possible to not block the bot from
+    /// handling events!
```

Comment:
```suggestion
    /// Checks if the `event` should be dispatched (`true`) or ignored (`false`).
    ///
    /// This affects [`crate::collector::collect`], [`crate::framework::Framework::dispatch`] and this `EventHandler` trait.
    ///
    /// ## Warning
    ///
    /// This will run synchronously on every event in the dispatch loop
    /// of the shard that is receiving the event. If your filter code
    /// takes too long, it may delay other events from being dispatched
    /// in a timely manner. It is recommended to keep the runtime
    /// complexity of the filter code low to avoid unnecessarily blocking
    /// your bot.
```